panda/src/glstuff/glCgShaderContext_src.cxx

```diff
@@ -876,48 +878,50 @@ update_shader_vertex_arrays(ShaderContext *prev, bool force) {
         } else {
           // It's a conventional vertex attribute.  Ugh.
 #ifdef SUPPORT_FIXED_FUNCTION
-          switch (p) {
-          case CA_unknown:
-            break;
-
-          case CA_vertex:
-            glVertexPointer(num_values, type, stride, client_pointer);
-            glEnableClientState(GL_VERTEX_ARRAY);
-            break;
-
-          case CA_normal:
-            glNormalPointer(type, stride, client_pointer);
-            glEnableClientState(GL_NORMAL_ARRAY);
-            break;
-
-          case CA_color:
-            if (numeric_type == GeomEnums::NT_packed_dabc) {
-              glColorPointer(GL_BGRA, GL_UNSIGNED_BYTE, stride, client_pointer);
-            } else {
-              glColorPointer(num_values, type, stride, client_pointer);
-            }
-            glEnableClientState(GL_COLOR_ARRAY);
-            break;
-
-          case CA_secondary_color:
-            _glgsg->_glSecondaryColorPointer(num_values, type,
-                                             stride, client_pointer);
-            glEnableClientState(GL_SECONDARY_COLOR_ARRAY);
-            break;
+          if(_glgsg->has_fixed_function_pipeline()) {
```

Comment:
I think this check doesn't need to be here.  We never need to set it to `CA_*` if `has_fixed_function_pipeline()` is false anyway.